File: algorithms/1710.maximum-units-on-a-truck/solution.go
func maximumUnits(boxTypes [][]int, truckSize int) int {
	res := 0

	sort.Slice(boxTypes, func(i, j int) bool {
		return boxTypes[i][1] > boxTypes[j][1]
	})

	for _, boxType := range boxTypes {
		num, unit := boxType[0], boxType[1]

		if num > truckSize {
			res += truckSize * unit
			break
		} else {
			res += num * unit
			truckSize -= num
		}
	}

	return res
}

File: algorithms/1710.maximum-units-on-a-truck/solution.rs
impl Solution {
    pub fn maximum_units(box_types: Vec<Vec<i32>>, truck_size: i32) -> i32 {
        let mut res = 0;

        let mut box_types: Vec<Vec<i32>> = box_types;
        box_types.sort_by_key(|x| -x[1]);

        let mut truck_size = truck_size;
        for box_type in box_types {
            let (num, unit) = (box_type[0], box_type[1]);

            if num > truck_size {
                res += truck_size * unit;
                break;
            } else {
                res += num * unit;
                truck_size -= num
            }
        }

        res
    }
}
